### PR TITLE
feat(v5): Phase 4 T4b — RemoteMediaClient TS façade + hooks

### PR DESCRIPTION
**Pull request by petrbela**
## Phase 4 — T4b: RemoteMediaClient TS façade + hooks (`v5-aug.2`)

Thin TS façade over the singleton transport, mirroring the `CastSession` generation-guard pattern: a handle retained across a disconnect is **stale** — mutations reject `noSession` before crossing the bridge, reads return `null` rather than leak a later session's status (Invariant 3, media flavour).

- `RemoteMediaClient`: all 19 transport mutations + 2 sync reads (`getMediaStatus`/`getStreamPosition`); per-store `WeakMap` memo keyed by generation.
- Hooks: `useRemoteMediaClient` / `useMediaStatus` / `useStreamPosition` via `useSyncExternalStore`.
- Reads served synchronously from the central store's media slice; mutations route through the T4a transport surface.
- Public API + media types exported from `src/index.ts`.

**Tests:** 34 new (façade routing/guard/factory via `FakeCastTransport` + hook render wiring). Full suite **92 green**, `yarn typescript` clean. Transport untouched.

**v4-parity reconcile notes:** `customData` survives only on `loadMedia`; listeners → hooks; `requestStatus` → `requestMediaStatus`; v4 convenience methods dropped. Tracked on `v5-aug.2`.

Stacked on the T4a foundation (already on `v5`). Disjoint from the iOS/Android PRs — independently mergeable.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a remote media client API for playback control, seeking, track/volume management, queue actions, and status refresh.
  * Introduced React hooks for the active remote media client, current media status, and stream position.
  * Expanded public exports with additional remote-media types.
* **Bug Fixes**
  * Improved stale-session behavior: reads return `null`, and outdated actions are rejected.
  * Prevented media status from reappearing across session changes, including additional lifecycle teardown paths.
* **Tests**
  * Added/updated unit and hook tests covering client memoization, lifecycle edge cases, defaults, and store update wiring.

### CODE REVIEW

File: src/api/useMediaStatus.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * `mediaStatus.streamPosition` reflects the time of the last update, not the
+ * live position — use {@link useStreamPosition} for the latter. Returns `null`
+ * when there is no active media (no session, or none loaded yet).
+ *
+ * @returns the current media status, or `null`.
+ */
+export function useMediaStatus(): MediaStatus | null {
+  return useSyncExternalStore(
+    castStore.subscribe,
+    () => castStore.getSliceState<MediaState>(MEDIA_SLICE_KEY).currentStatus
```
[the remark this answers is not in the document]
Comment (petrbela) [reply src/api/useMediaStatus.ts:20]: Good catch — this was a real null-on-no-session leak. Root cause was two hand-maintained event lists drifting: `session.slice` tears down the live session on four events (`ended`, `startFailed`, `resumeFailed`, `suspended`), but the media slice only cleared on the first two, so the Android `suspended` / `resumeFailed` background paths left stale media behind `useMediaStatus` / `useStreamPosition` after `useRemoteMediaClient()` had gone null.

Fixed in 6c829f6 by exporting `SESSION_TEARDOWN_TYPES` from `session.slice` and clearing the media slice on that shared set (rather than re-listing the subset, which would just re-introduce the drift). Added `suspended` and `resumeFailed` clearing tests plus an extended stable-reference invariant.

File: src/api/useMediaStatus.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { MEDIA_SLICE_KEY, type MediaState } from '../state/media.slice'
+import type { MediaStatus } from '../types/MediaStatus'
+
+/**
+ * Hook to retrieve the current media status.
+ *
+ * The status is pushed only when the stream's status changes, so
+ * `mediaStatus.streamPosition` reflects the time of the last update, not the
+ * live position — use {@link useStreamPosition} for the latter. Returns `null`
+ * when there is no active media (no session, or none loaded yet).
```
[the remark this answers is not in the document]
Comment (petrbela) [reply src/api/useMediaStatus.ts]: Reworded in 6c829f6 — the JSDoc now describes `useStreamPosition()` as a convenience selector for the `streamPosition` field rather than a live/ticking source, matching its own documented status-driven behavior.

File: src/api/useRemoteMediaClient.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * @returns the current client, or `null` if there is no session connected.
+ *
+ * @example
+ * ```ts
+ * import { useRemoteMediaClient } from 'react-native-google-cast'
+ *
+ * function MyComponent() {
+ *   const client = useRemoteMediaClient()
+ *   client?.loadMedia({ mediaInfo: { contentUrl: '…' } })
+ * }
```
[the remark this answers is not in the document]
Comment (petrbela) [reply src/api/useRemoteMediaClient.ts:25]: Fixed in 6c829f6 — moved the `loadMedia` call into a `useEffect` keyed on `client` (per the suggested doc fix) so the example no longer issues a load on every render.

File: src/api/RemoteMediaClient.ts [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+      throw error
+    }
+  }
+
+  // --- reads (synchronous; served from the media slice cache) ---
+
+  /** The current media status, or `null` if there is no media (or stale handle). */
+  getMediaStatus(): MediaStatus | null {
+    if (!this.isActive) return null
+    return this.store.getSliceState<MediaState>(MEDIA_SLICE_KEY).currentStatus
```
[the remark this answers is not in the document]
Comment (petrbela) [reply src/api/RemoteMediaClient.ts:114]: Valid — fixed in 2f4652a by clearing the media slice on session **establishment** as well, so a replacement `started`/`resumed` resets `currentStatus` and the new generation's client (and the hooks) read `null` until the receiver pushes a fresh status, rather than briefly exposing the prior session's media. Part of a broader redesign that gates media-status application on live-session presence (see the media.slice thread). Added a replacement-clears test.

File: src/state/media.slice.ts [hunk 5]
```diff
@@ -32,9 +35,11 @@ export const mediaSlice: Slice<MediaState> = {
...
     }
     if (event.kind === 'lifecycle') {
-      // A status without a live session is a use-after-free waiting to happen.
-      const { type } = event.event
-      if (type === 'ended' || type === 'startFailed') {
+      // A status without a live session is a use-after-free waiting to happen,
+      // so clear on every teardown the session slice recognises — not a
+      // hand-copied subset that can silently drift (the Android `suspended` /
+      // `resumeFailed` paths drop the session too).
+      if (SESSION_TEARDOWN_TYPES.has(event.event.type)) {
```
[the remark this answers is not in the document]
Comment (petrbela) [reply src/state/media.slice.ts]: Good catch — fixed in 2f4652a. Rather than relying on a later teardown to clean up, the media slice now tracks session liveness directly: it carries a `live` bit mirroring `session.current !== null`, derived from the **same** shared lifecycle event sets (`SESSION_ESTABLISH_TYPES` / `SESSION_TEARDOWN_TYPES`) and the same "establish carries a session" rule the session slice uses, so the two can't drift. A `mediaStatus` push is now **applied only while `live`**, so `started → suspended → mediaStatus` no longer repopulates `currentStatus`. The suspended/resumeFailed tests now emit that later push and assert it's dropped (the path you flagged as unverified).
